lua/gatekeeper/staticlib.lua

```diff
@@ -124,6 +124,41 @@ function convert_numa_table_to_array (numa_table)
 	return res
 end
 
+function gk_sol_map (gk_lcores, sol_lcores)
+	local m = {}
+	local d = math.ceil(#gk_lcores / #sol_lcores)
+	local sol_remaining_cap = {}
+
+	if #gk_lcores % #sol_lcores ~= 0 then
+		print("Wrong GK-to-SOL assignment");
+	end
+
+	for i, v in ipairs(sol_lcores) do
+		table.insert(sol_remaining_cap, d)
+	end
+
+	for i1, v1 in ipairs(gk_lcores) do
+		local idx
+		local socket_id = rte_lcore_to_socket_id(v1)
+
+		for i2, v2 in ipairs(sol_lcores) do
+			if rte_lcore_to_socket_id(v2) == socket_id and
+					(idx == nil or sol_remaining_cap[i2] >
+						sol_remaining[idx]) then
+				idx = i2
+			end
+		end
+
+		if idx == nil or sol_remaining_cap[idx] <= 0 then
+			error("Wrong GK-to-SOL assignment")
```

Comment:
Once the changes discussed in the other comments are implemented, the test should be only `idx == nil`, and the error message `"No Sol block allocated at NUMA node " .. socket_id`.